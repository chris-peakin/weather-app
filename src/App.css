.App {
}

.App-logo {

}

@media {

}

.App-header {

}

.App-link {

}

@keyframes App-logo-spin {

}
