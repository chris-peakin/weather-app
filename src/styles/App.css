.weather-app {
    margin: 0 20px;
}

.App-logo {

}

@media {

}

.App-header {

}

.App-link {

}

@keyframes App-logo-spin {

}
