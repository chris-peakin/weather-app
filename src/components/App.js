import "../styles/App.css";
import React, { useEffect, useState } from "react";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "./getForecast";
import SearchForm from "./SearchForm";

function App() {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("Manchester");
  const [errorMessage, setErrorMessage] = useState("");

  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  useEffect(() => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);

  return (
    <div className="forecast">
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        onSubmit={handleCitySearch}
        searchText={searchText}
        setSearchText={setSearchText}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails details={selectedForecast} />}
        </>
      )}
    </div>
  );
}

export default App;
