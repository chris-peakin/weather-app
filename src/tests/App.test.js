import { render, screen } from "@testing-library/react";
import App from "../components/App";
import forecast from "../data/forecast.json";

describe("App", () => {
  test("renders App correctly after pulling live data", () => {
    const {asFragment} = render (<App />)
    expect(asFragment()).toMatchSnapshot();
  });
});