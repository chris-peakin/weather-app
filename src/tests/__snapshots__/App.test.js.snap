// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly after pulling live data 1`] = `
<DocumentFragment>
  <div
    class="forecast"
  >
    <h1
      class="location-details"
    >
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
